cmd/sourced/cmd/root.go

```diff
@@ -42,14 +42,31 @@ type Command struct {
 // Execute adds all child commands to the root command and sets flags appropriately.
 // This is called by main.main(). It only needs to happen once to the rootCmd.
 func Execute() {
-	if err := rootCmd.Run(os.Args); err != nil {
-		if workdir.ErrMalformed.Is(err) || dir.ErrNotExist.Is(err) {
-			fmt.Println(format.Colorize(
-				format.Red,
-				`Cannot perform this action, source{d} needs to be initialized first with the 'init' sub command`,
-			))
-		}
+	if err := dir.Prepare(); err != nil {
+		fmt.Println(err)
+		log(err)
+		os.Exit(1)
+	}
 
+	if err := rootCmd.Run(os.Args); err != nil {
+		log(err)
 		os.Exit(1)
 	}
 }
+
+func log(err error) {
+	switch {
+	case workdir.ErrMalformed.Is(err) || dir.ErrNotExist.Is(err):
+		printRed("Cannot perform this action, source{d} needs to be initialized first with the 'init' sub command")
+	case dir.ErrNotValid.Is(err):
+		printRed("Cannot perform this action, config directory is not valid")
+	case fmt.Sprintf("%T", err) == "*flags.Error":
+		// syntax error is already logged by go-cli
+	default:
+		printRed("Unexpected error")
```

Comment:
Isn't this redundant? I would not print anything in this case, it does not add anything useful for the user
```
$ sudo mkdir /tmp/aaa
$ HOME=/tmp/aaa sourced status
mkdir /tmp/aaa/.sourced: permission denied
Unexpected error
```

Reply:
It was to provide homogeneous error messages, but I assume we can also remove it if it's unexpected;
Anyhow, that should not be "unexpected", will review what happened in that case.

Reply:
https://github.com/src-d/sourced-ce/pull/162/commits/78f80cd7e51ff00f6ebdfeee3800c51085e656f8 fixes that @carlosms
- unknown errors will have no special message but the causing error itself,
- fixed that example, that should be as it follows:
```shell
$ HOME=/root/User sourced status
/root/User/.sourced is not a valid config directory: mkdir /root/User/.sourced: permission denied
Cannot perform this action, config directory is not valid
```